packages/react-instantsearch/src/core/highlight.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+
+function extractTagName(highlightedValue) {
+  const tagParse = (/<(.+?)>/).exec(highlightedValue);
+  return tagParse === null ? null : tagParse[1];
+}
+
+/**
+ * This function transforms an highlighted response
+ * to an array of React elements. This should be
+ * used in place of dangerouslySetInnerHTML when
+ * using the highlightReasult provided by Algolia.
+ *
+ * The function autodetects which tag is used in
+ * results to highlight, and creates the corresponding
+ * React element.
+ *
+ * It also creates keys on React elements to avoid warnings.
+ *
+ * @param {string} attribute the name of the attribute for which we want the highlighted value
+ * @param {object} hit the hit from the algolia response
+ * @returns {Array.<string|ReactElement>} the array of react element
+ * @example
+ * const ProductHits = Hits.connect(({hits}) => {
+ *   const hitComponents = hits.map( hit =>
+ *     <div key={hit.objectID}>
+ *       <span className="hit-name">{hightlight('name', hit)}</span>
+ *     </div>
+ *   );
+ *
+ *   return <div className="hits">{hitComponents}</div>;
+ * });
+ */
+export default function hightlight(attribute, hit) {
+  if (!hit) throw new Error('The hit containing the attribute should be provided');
+  if (!{}.hasOwnProperty.call(hit, attribute)) throw new Error(`${attribute} should be a retrievable attribute.`);
```

Comment:
Just read the documentation on eslint, we are find disabling this rule as a whole in the .eslintrc.js. We will then backport it in eslint-config-algolia. The reasoning for the rule is not worth the weird syntax they provide as an alternative to me.